aws-ssm-parameter/src/main/java/com/amazonaws/ssm/parameter/CreateHandler.java

```diff
@@ -0,0 +1,106 @@
+package com.amazonaws.ssm.parameter;
+
+import org.apache.commons.lang3.RandomStringUtils;
+
+import software.amazon.awssdk.services.ssm.SsmClient;
+import software.amazon.awssdk.services.ssm.model.ParameterNotFoundException;
+import software.amazon.awssdk.services.ssm.model.PutParameterRequest;
+import software.amazon.awssdk.services.ssm.model.PutParameterResponse;
+import software.amazon.awssdk.services.ssm.model.TooManyUpdatesException;
+import software.amazon.cloudformation.exceptions.CfnNotFoundException;
+import software.amazon.cloudformation.exceptions.CfnThrottlingException;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+import software.amazon.cloudformation.proxy.Logger;
+import software.amazon.cloudformation.proxy.ProxyClient;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+import software.amazon.cloudformation.proxy.delay.Constant;
+
+import java.time.Duration;
+import java.util.Map;
+import java.util.Random;
+
+public class CreateHandler extends BaseHandlerStd {
+    private static final String OPERATION = "PutParameter";
+    private static final String RETRY_MESSAGE = "Detected retryable error, retrying. Exception message: %s";
+    private Logger logger;
+
+    @Override
+    protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+            final AmazonWebServicesClientProxy proxy,
+            final ResourceHandlerRequest<ResourceModel> request,
+            final CallbackContext callbackContext,
+            final ProxyClient<SsmClient> proxyClient,
+            final Logger logger) {
+        this.logger = logger;
+        final ResourceModel model = request.getDesiredResourceState();
+
+        Map<String, String> consolidatedTagList = request.getDesiredResourceTags();
+        consolidatedTagList.putAll(request.getSystemTags());
+
+        // Set model primary ID if absent
+        if(model.getName() == null) {
+            model.setName(generateParameterName(
+                    request.getLogicalResourceIdentifier(),
+                    request.getClientRequestToken()
+            ));
+        }
+
+        Constant BACK_OFF_DELAY;
+
+        if(model.getDataType() != null && model.getDataType() == Constants.AWS_EC2_IMAGE_DATATYPE) {
+            BACK_OFF_DELAY = Constant.of()
+                    .timeout(Duration.ofMinutes(5))
+                    .delay(Duration.ofSeconds(30))
+                    .build();
+        } else {
+            BACK_OFF_DELAY = Constant.of()
+                    .timeout(Duration.ofMinutes(5))
+                    .delay(Duration.ofSeconds(5))
+                    .build();
+        }
```

Comment:
same thing is used in `UpdateHandler`? maybe move it to `BaseHandler`

Reply:
Fixed